extensions/standard-processors/processors/LogAttribute.cpp

```diff
@@ -137,33 +137,32 @@ void LogAttribute::onTrigger(const std::shared_ptr<core::ProcessContext> &contex
       message << "\n" << "Payload:" << "\n";
       ReadCallback callback(flow->getSize());
       session->read(flow, &callback);
-      for (unsigned int i = 0, j = 0; i < callback.read_size_; i++) {
-        message << std::hex << callback.buffer_[i];
-        j++;
-        if (j == 80) {
-          message << '\n';
-          j = 0;
-        }
+
+      auto payload_hex = utils::StringUtils::to_hex(callback.buffer_.data(), callback.buffer_.size());
+      for (size_t i = 0; i < payload_hex.size(); i += 80) {
+        message << payload_hex.substr(i, 80) << '\n';
       }
+    } else {
+      message << "\n";
     }
-    message << "\n" << dashLine;
+    message << dashLine;
     std::string output = message.str();
 
     switch (level) {
       case LogAttrLevelInfo:
-        logger_->log_info("%s", output);
```

Comment:
let's revert these.

Reply:
As I wrote in the global comment, this is a fix for the truncation issue. An another way to solve that would be to either enlarge the log format buffer size, or use a dynamic reallocation approach in it, but this seemed a less disruptive way to fix the bug.

Reply:
okay I'll take a look at that global comment a little closer later. Thanks for taking the time to write that!